pkg/loader/executor.go

```diff
@@ -118,19 +127,72 @@ func (tx *tx) commit() error {
 }
 
 func (e *executor) updateMark(channel string, tx *tx) error {
+	if e.info == nil {
+		return nil
+	}
+	index = e.getIndex(index)
```

Comment:
may have more than one thread use `index`, how about using `atomic.StoreInt64` and `atomic.LoadInt64` to set or get `index`?